android/src/main/java/com/amplitude/android/plugins/AndroidContextPlugin.kt

```diff
@@ -52,7 +56,18 @@ class AndroidContextPlugin : Plugin {
     private fun applyContextData(event: BaseEvent) {
         val configuration = amplitude.configuration as Configuration
         event.timestamp ?: let {
-            event.timestamp = System.currentTimeMillis()
+            val eventTime = System.currentTimeMillis()
+            event.timestamp = eventTime
+            (amplitude as com.amplitude.android.Amplitude).lastEventTime = eventTime
```

Comment:
Maybe we can make a getter to avoid frequent type casting?

```
class AndroidPluginBase: Plugin {
  getAmplitude(): com.amplitude.android.Amplitude {
   return amplitude as com.amplitude.android.Amplitude
  }
}
```

Reply:
I see, basically have the typecasting inside the plugin. That should be good for our own plugins and I can make that change. I wonder if in the future customer would need to do that typecasting in their plugins, and if we should switch to extension before that, or if they can do similar getter just for android.